### ISSUE TITLE
attributes_escape/1 does not escape ids completely

### ISSUE THREAD

**Issue by ponychicken** [phoenixframework/phoenix_live_view#3465]
I have data from a legacy system. On a few entries in noticed a subtle breakage. I traced it down to attributes_escape/1 not escaping ids sufficiently-

The docs hint that passing a number will throw an error:
```
iex(8)> attributes_escape(%{id: 11})
** (ArgumentError) attempting to set id attribute to 11, but setting the DOM ID to a number can lead to unpredictable behaviour. Instead consider prefixing the id with a string, such as "user-11" or similar
    (phoenix_html 4.1.1) lib/phoenix_html.ex:258: Phoenix.HTML.id_value/1
    (phoenix_html 4.1.1) lib/phoenix_html.ex:208: Phoenix.HTML.build_attrs/1
    (phoenix_html 4.1.1) lib/phoenix_html.ex:195: Phoenix.HTML.attributes_escape/1
    iex:8: (file)
```

But passing the number a string will happily produce output:
```
iex(7)> attributes_escape(%{id: "11"})
{:safe, [" id=\"", "11", 34]}
```

Same for whitespace ("An ID attribute's value must not contain [ASCII whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace#in_html) characters.")
```
iex(8)> attributes_escape(%{id: "white space"})
{:safe, [" id=\"", "white space", 34]}
```

Additionally also characters like "/" are not valid in ids, but the escape function will not escape them as well:
```
iex(4)> attributes_escape(%{id: "asdflkj//sdf"})
{:safe, [" id=\"", "asdflkj//sdf", 34]}
```

> Technically, the value for an ID attribute may contain any other Unicode character. However, when used in CSS selectors, either from JavaScript using APIs like [Document.querySelector()](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) or in CSS stylesheets, ID attribute values must be valid [CSS identifiers](https://developer.mozilla.org/en-US/docs/Web/CSS/ident). This means that if an ID attribute value is not a valid CSS identifier (for example, my?id or 1234) then it must be escaped before being used in a selector, either using the [CSS.escape()](https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape_static) method or [manually](https://developer.mozilla.org/en-US/docs/Web/CSS/ident#escaping_characters).

**Comment by SteffenDE** [comment]
@ponychicken can you provide an example to reproduce the error in LV? I just tried a button with id "foo/bar" and it worked just fine:

```heex
<button phx-click="inc">+</button>
<button id="foo/bar" phx-click="dec">-</button>
```

**Comment by ponychicken** [comment]
Uncaught SyntaxError: Failed to execute 'querySelectorAll' on 'Document': '#items-foo/bar-menu' is not a valid selector.
    at Object.all (dom.js:40:33)
    at Object.filterToEls (js.js:309:21)
    at js.js:18:12
    at Array.forEach (<anonymous>)
    at Object.exec (js.js:13:14)
    at live_socket.js:678:14
    at live_socket.js:467:33
    at LiveSocket.owner (live_socket.js:463:15)
    at LiveSocket.withinOwners (live_socket.js:467:10)
    at live_socket.js:677:14

**Comment by josevalim** [comment]
@ponychicken what is your LV version? It may have been fixed in more recent ones.

**Comment by ponychicken** [comment]
I was on rc6 and just upgraded to rc7 with the same result

**Comment by josevalim** [comment]
@SteffenDE maybe we should make css_escape part of Phoenix.HTML? Going back to the starting point here. :)

**Comment by SteffenDE** [comment]
@ponychicken this will be part of the next `:phoenix_html` release, as well as the next Floki release. Together with the already mentioned library (https://github.com/SteffenDE/css_escape), you'll have plenty of options :)

I hope that this helps!

**Comment by ponychicken** [comment]
@SteffenDE  Thank you <3333

**Comment by SteffenDE** [comment]
@ponychicken this is now released as part of phoenix_html 4.2 as well as Floki 0.37 :)
